tgui/packages/tgui/interfaces/Smes.js

```diff
@@ -0,0 +1,188 @@
+import { useBackend } from '../backend';
+import { Box, Button, Flex, LabeledList, ProgressBar, Section, Slider } from '../components';
+import { formatPower } from '../format';
+import { Window } from '../layouts';
+
+// Common power multiplier
+const POWER_MUL = 1e3;
+
+export const Smes = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    capacityPercent,
+    capacity,
+    charge,
+    inputAttempt,
+    inputting,
+    inputLevel,
+    inputLevelMax,
+    inputAvailable,
+    outputAttempt,
+    outputting,
+    outputLevel,
+    outputLevelMax,
+    outputUsed,
+  } = data;
+  const inputState = (
+    capacityPercent >= 100 && 'good'
+    || inputting && 'average'
+    || 'bad'
+  );
+  const outputState = (
+    outputting && 'good'
+    || charge > 0 && 'average'
+    || 'bad'
+  );
+  return (
+    <Window>
+      <Window.Content>
+        <Section title="Stored Energy">
+          <ProgressBar
+            value={capacityPercent * 0.01}
+            ranges={{
+              good: [0.5, Infinity],
+              average: [0.15, 0.5],
+              bad: [-Infinity, 0.15],
+            }} />
+        </Section>
+        <Section title="Input">
+          <LabeledList>
+            <LabeledList.Item
+              label="Charge Mode"
+              buttons={
+                <Button
+                  icon={inputAttempt ? 'sync-alt' : 'times'}
+                  selected={inputAttempt}
+                  onClick={() => act('tryinput')}>
+                  {inputAttempt ? 'Auto' : 'Off'}
+                </Button>
+              }>
+              <Box color={inputState}>
+                {capacityPercent >= 100 && 'Fully Charged'
+                  || inputting && 'Charging'
+                  || 'Not Charging'}
+              </Box>
+            </LabeledList.Item>
+            <LabeledList.Item label="Target Input">
+              <Flex inline width="100%">
+                <Flex.Item>
+                  <Button
+                    icon="fast-backward"
+                    disabled={inputLevel === 0}
+                    onClick={() => act('input', {
+                      target: 'min',
+                    })} />
+                  <Button
+                    icon="backward"
+                    disabled={inputLevel === 0}
+                    onClick={() => act('input', {
+                      adjust: -10000,
+                    })} />
+                </Flex.Item>
+                <Flex.Item grow={1} mx={1}>
+                  <Slider
+                    value={inputLevel / POWER_MUL}
+                    fillValue={inputAvailable / POWER_MUL}
+                    minValue={0}
+                    maxValue={inputLevelMax / POWER_MUL}
+                    step={5}
+                    stepPixelSize={4}
+                    format={value => formatPower(value * POWER_MUL, 1)}
+                    onDrag={(e, value) => act('input', {
```

Comment:
`onChange` here so people cant overload the server with mass hrefs by dragging the slider up and down without letting go.